### PR TITLE
email fix for score_plugins workflow

### PR DESCRIPTION
**Pull request by samwinebrake**
Related to closed PR #579 

Broke that PR up into two parts:

1. (THIS PR) a fix for the current score_plugins workflow that finds brain-score emails based on uid and funnels these emails to jenkins for correct email reporting of submission success/failure
2. (FOR A FUTURE PR) failure emails for failing unit testing jenkins jobs

Everything is tested and working, as long as secrets.AWS_ACCESS_KEY_ID and secrets.AWS_SECRET_ACCESS_KEY are set within vision.

(See previous comments from [here](https://github.com/brain-score/vision/pull/579) for input on choices made in this PR.)

### CODE REVIEW

File: .github/workflows/score_new_plugins.yml [hunk 1]
```diff
@@ -20,12 +20,12 @@ permissions: write-all
 
 jobs:
 
-  changes_models_or_benchmarks:
+  process_submission:
     name: Check if PR makes changes to /models or /benchmarks
```
Comment (kvfairchild) [inline .github/workflows/score_new_plugins.yml:24]:
```suggestion
    name: If triggering PR alters /models or /benchmarks, initiates scoring for relevant plugins
```

File: .github/workflows/score_new_plugins.yml [hunk 2]
```diff
@@ -53,50 +60,49 @@ jobs:
...
+      - name: Update PLUGIN_INFO for automerge-web (find uid, public v. private, and bs email)
+        if: contains(github.event.pull_request.labels.*.name, 'automerge-web') && steps.scoringneeded.outputs.RUN_SCORING == 'True'
+        id: automerge_web
         run: |
           BS_UID="$(echo '${{ github.event.pull_request.title }}' | sed -E 's/.*\(user:([^)]+)\).*/\1/')"
           BS_PUBLIC="$(echo '${{ github.event.pull_request.title }}' | sed -E 's/.*\(public:([^)]+)\).*/\1/')"
-          echo "The Brain-Score user ID is $BS_UID"
-          echo "PLUGIN_INFO=$(<<<$PLUGIN_INFO tr -d "'" | jq -c ". + {user_id: \"$BS_UID\", public: \"$BS_PUBLIC\"}")" >> $GITHUB_ENV
+          USER_EMAIL=$(python -c "from brainscore_core.submission.database import email_from_uid; from brainscore_core.submission.endpoints import UserManager; user_manager=UserManager(db_secret='${{ secrets.BSC_DATABASESECRET }}'); print(email_from_uid($BS_UID))")
+          echo "::add-mask::$USER_EMAIL" # Mask the USER_EMAIL
```
Comment (kvfairchild) [inline .github/workflows/score_new_plugins.yml:90]: 😎 excellent InfoSec

File: .github/workflows/score_new_plugins.yml [hunk 3]
```diff
@@ -53,50 +60,49 @@ jobs:
...
-          github-username: ${{github.event.pull_request.user.login}} # PR author's username
-          token: ${{ secrets.GITHUB_TOKEN }} # Including token enables most reliable way to get a user's email
-      - name: Add PR author email to PLUGIN_INFO
-        id: add_email_to_pluginfo
+      - name: Set job-level output for PLUGIN_INFO
+        id: set_plugin_output
         run: |
-          echo "The PR author email is ${{ steps.getemail.outputs.email }}"
-          echo "PLUGIN_INFO=$(<<<$PLUGIN_INFO tr -d "'"  | jq -c '. + {author_email: "${{ steps.getemail.outputs.email }}"}')" >> $GITHUB_OUTPUT
+          echo "::set-output name=PLUGIN_INFO::$PLUGIN_INFO"
```
Comment (kvfairchild) [inline .github/workflows/score_new_plugins.yml:96]: `set-output` is [set to be deprecated](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/); this should be (I think; test before merging!)
```suggestion
          echo "{PLUGIN_INFO}={$PLUGIN_INFO}" >> $GITHUB_OUTPUT
```

**Review by kvfairchild — APPROVED** [review]
Some changes requested but (as long as this is fully tested), looks good to me.

As we're adding logic for notifying the submitter on failure across multiple actions, you may want to look into [reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) to centralize the logic and reduce duplication.

**Comment by samwinebrake** [comment]
> Some changes requested but (as long as this is fully tested), looks good to me.
> 
> As we're adding logic for notifying the submitter on failure across multiple actions, you may want to look into [reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) to centralize the logic and reduce duplication.

Good point! I'll merge this for now so we have emails going out to users and prioritize creating a reusable user-notifier action that can be used for this and other CI results.

**Pull request merged**